docs/contributing/development_environment.md

```diff
@@ -170,66 +188,354 @@ network:
   DockerInsecureRegistryAddress:
-    - $CTLPLANE_IP:8787
+  - $CTLPLANE_IP:8787
   NeutronPublicInterface: $NEUTRON_INTERFACE
-  NtpServer: $NTP_SERVER
-  CloudDomain: localdomain
+  # domain name used by the host
   NeutronDnsDomain: localdomain
-  NeutronBridgeMappings: datacentre:br-ctlplane
-  NeutronPhysicalBridge: br-ctlplane
+  # re-use ctlplane bridge for public net
+  NeutronBridgeMappings: datacentre:$BRIDGE
+  NeutronPhysicalBridge: $BRIDGE
   StandaloneEnableRoutedNetworks: false
   StandaloneHomeDir: $HOME
   InterfaceLocalMtu: 1500
+  # Needed if running in a VM
   NovaComputeLibvirtType: qemu
+  ValidateGatewaysIcmp: false
+  ValidateControllersIcmp: false
 EOF
 
-openstack tripleo container image prepare default \
-  --output-env-file $HOME/containers-prepare-parameters.yaml
-
-cat << EOF > $HOME/standalone.sh
-set -euxo pipefail
-
-if podman ps | grep keystone; then
-    echo "Looks like OpenStack is already deployed, not re-deploying."
-    exit 0
-fi
-
 sudo openstack tripleo deploy \
-  --templates \
-  --local-ip=$CTLPLANE_IP/24 \
-  --control-virtual-ip=$CTLPLANE_VIP \
+  --templates /usr/share/openstack-tripleo-heat-templates \
+  --standalone-role Standalone \
   -e /usr/share/openstack-tripleo-heat-templates/environments/standalone/standalone-tripleo.yaml \
-  -r /usr/share/openstack-tripleo-heat-templates/roles/Standalone.yaml \
-  -e "$HOME/containers-prepare-parameters.yaml" \
-  -e "$HOME/standalone_parameters.yaml" \
   -e /usr/share/openstack-tripleo-heat-templates/environments/low-memory-usage.yaml \
+  -e ~/containers-prepare-parameters.yaml \
+  -e standalone_parameters.yaml \
+  -e /usr/share/openstack-tripleo-heat-templates/environments/cephadm/cephadm.yaml \
+  -e ~/deployed_ceph.yaml \
+  -e /usr/share/openstack-tripleo-heat-templates/environments/deployed-network-environment.yaml \
+  -e deployed_network.yaml \
+  -r /usr/share/openstack-tripleo-heat-templates/roles/Standalone.yaml \
+  -n network_data.yaml \
+  --local-ip=$CTLPLANE_IP/$CIDR \
+  --control-virtual-ip=$CTLPLANE_VIP \
   --output-dir $HOME
+```
+
+### Create a workload to adopt
+
+For this example we'll upload a Glance image and confirm it's using
+the Ceph cluster. Later you can adopt the Glance image in the NG
+deployment.
 
+Download a cirros image and convert it to raw format for Ceph.
+```
+IMG=cirros-0.5.2-x86_64-disk.img
+URL=http://download.cirros-cloud.net/0.5.2/$IMG
+RAW=$(echo $IMG | sed s/img/raw/g)
+curl -L -# $URL > $IMG
+qemu-img convert -f qcow2 -O raw $IMG $RAW
+```
+Upload the image to Glance.
+```
 export OS_CLOUD=standalone
-openstack endpoint list
+openstack image create cirros --disk-format=raw --container-format=bare < $RAW
+```
+Confirm the image UUID can be seen in Ceph's images pool.
+```
+sudo cephadm shell -- rbd -p images ls -l
+```
+
+## Complete the NG Deployment
+
+The remaining steps should be completed on the hypervisor hosting crc
+and edpm-compute-0.
+
+### Deploy NG Control Plane with Ceph
```

Comment:
Not sure about this: I think services should be connected to the `Ceph` cluster as long as they're adopted, so if you follow the glance adoption doc there are steps (including the secrets creation) to run the service and connecting it to the external Ceph.
Shouldn't be the ctlplane almost empty at this stage? /cc @jistr 
Anyway the change itself is good, just wanted to clarify this point before merging.

Reply:
Yes i think the dev env setup should produce an environment with "old" OpenStack (+ Ceph) and an empty OpenShift/CRC. Further steps should probably be part of the actual Adoption. (In the "new design" we don't expect to have OpenStack control plane services running at the same time on old and new.)

That said i think the PR looks good and it's great that it worked for Fabricio (i'm still yet to test manually). So i'd say we should merge and iterate from there.

Reply:
I agree we want the new control plane empty. 

I include these steps since I wanted to show people how easy it is to connect the NG system to the old ceph cluster. How to extract the ceph conf/keys and create the secret. I don't know if that part is as obvious to someone who doesn't use ceph regularly.

I then emphasized that they shouldn't be confused about two services writing to the same ceph and thus they might want to test the connection but that they shouldn't keep the testing data or just use the NG system. Line 436 explains this and then have them delete the test data so that the new control plane remains empty. I then point out that this is a part of the glance adoption and link to your document.

Thanks all for the revews!